src/main/java/com/zaxxer/hikari/pool/PoolBase.java

```diff
@@ -313,7 +313,6 @@ private Connection newConnection() throws Exception
 
          connection = (username == null) ? dataSource.getConnection() : dataSource.getConnection(username, password);
          setupConnection(connection);
-         lastConnectionFailure.set(null);
```

Comment:
@brettwooldridge : ...'I am prepared to merge this request, but not with this change.'
any correction required for merging this?